src/software/util/parameter/generate_parameters_v2.py

```diff
@@ -0,0 +1,359 @@
+        config this parameter will be in.
+
+        :returns: string that goes in the constructor of the parent config
+        :rtype: str
+
+        """
+        return constants.PARAMETER_CONSTRUCTOR_ENTRY.format(
+            # the type of the parameter
+            type=self.ptype,
+
+            # the variable name
+            param_variable_name=self.param_variable_name,
+
+            # we need to add additional quotes around the value if it is a string
+            # so the quotes are formatted here
+            quote="\"" if self.ptype == 'std::string' else "",
+
+            # the parameter name (different from variable name)
+            param_name=self.param_name,
+
+            # min and max
+            param_min=self.min_value,
+            param_max=self.max_value,
+
+            # the parameter takes in a vector of options
+            # which is generated here
+            options=','.join(('"{0}"' if self.ptype == 'std::string' else '{0}')
+                             .format(option) for option in self.options),
+
+            # the default value of the parameter
+            value=self.default_value)
+
+    def get_private_entries(self):
+        """Returns the public members of the parent Config class
+        that are required for this parameter
+
+        :returns: entries into the private section of the Config class
+        :rtype: str
+
+        """
+        return constants.PARAMETER_PRIVATE_ENTRY.format(
+            type=self.ptype,
+            param_variable_name=self.param_variable_name)
+
+    def get_public_entries(self):
+        """Returns the public members of the parent Config class
+        that are required for this parameter.
+
+        :returns: entries into the public section of the Config class
+        :rtype: str
+
+        """
+        return constants.PARAMETER_PUBLIC_ENTRY.format(
+            type=self.ptype,
+            immutable_accessor_name=''+self.param_name.strip(),
+            mutable_accessor_name='mutable'+self.param_name.strip(),
+            param_variable_name=self.param_variable_name)
+
+#######################################################################
+#                            CPP Generator                            #
+#######################################################################
+
+
+class Config(object):
+
+    """Recursive data structure representing the equivalent c++ class
+    in config.hpp that defines a configuration. Created from a yaml
+    dictionary that the user specifies.
+
+    :param config_name: the name of the config
+    :param config_description: the yaml configuration for the config 
+    :type config_name: str
+    :type config_description: dict
+
+    """
+
+    # stores all the configurations in a "flat" structure
+    # globally to print all the classes and their
+    # forward declarations
+    all_configs = []
+
+    def __init__(self, config_name: str,  config_description: dict):
+
+        """INITIALIZER"""
```

Comment:
Not sure this adds much value either, same goes for other instances?

Reply:
removed